src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -2319,7 +2319,7 @@ public static void Refresh(Assembly assembly)
         [EditorBrowsable(EditorBrowsableState.Advanced)]
         public static Type ComObjectType
         {
-            [DynamicallyAccessedMembersAttribute(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+            [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
```

Comment:
(nit) - this may be a preference thing, but IMO it feels more natural for this attribute to be applied to the property:

```C#
[EditorBrowsable(EditorBrowsableState.Advanced)]
[DynamicallyAccessedMembersAttribute(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
public static Type ComObjectType
{
    get => typeof(TypeDescriptorComObject);
}

Reply:
I'm actually not sure linker will be able to handle this - the attribute on a property works for implicit getters/setters, but for those with explicit bodies it might not be reliable (since we need to detect the backing field if there's one to annotate it as well, and that detection is tricky/impossible for random properties).

Reply:
Note that if it doesn't work linker will not warn - TODO - I created https://github.com/mono/linker/issues/1248 to get that fixed.

Reply:
> I'm actually not sure linker will be able to handle this - the attribute on a property works for implicit getters/setters, but for those with explicit bodies it might not be reliable (since we need to detect the backing field if there's one to annotate it as well, and that detection is tricky/impossible for random properties).

The behavior is that if the annotation is placed on the property, the annotation is propagated to the return value of the getter and the parameter of the setter. If this is an auto property, the annotation is also propagated to the backing field. Otherwise, this won't propagate to the backing field, even if it looks "obvious". One will get a warning and needs to annotate the backing field manually.

Reply:
What about this case where there is no backing field? Does the `TypeDescriptorComObject` type get annotated with `[DynamicallyAccessedMembersAttribute(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]` if the attribute is applied to the property?

Reply:
> What about this case where there is no backing field?

This part applies always: "The behavior is that if the annotation is placed on the property, the annotation is propagated to the return value of the getter and the parameter of the setter"

> Does the TypeDescriptorComObject type get annotated with

Yes, because it's a value that was returned from a method that has the return parameter annotated (because the annotation was propagated from the property).